flink/v1.20/flink/src/test/java/org/apache/iceberg/flink/data/TestFlinkParquetReader.java

```diff
@@ -236,4 +457,44 @@ protected void writeAndValidate(Schema schema) throws IOException {
         RandomGenericData.generateFallbackRecords(schema, NUM_RECORDS, 21124, NUM_RECORDS / 20),
         schema);
   }
+
+  @Override
+  protected void writeAndValidate(Schema writeSchema, Schema expectedSchema) throws IOException {
+    assumeThat(
+            TypeUtil.find(
+                writeSchema,
+                type -> type.isMapType() && type.asMapType().keyType() != Types.StringType.get()))
+        .as("Parquet Avro cannot write non-string map keys")
+        .isNull();
```

Comment:
Why is this using parquet-avro to read when the version of `writeAndValidate` just above does not have this restriction? There are assertions to validate `Record` with `RowData` so I think this should use the same approach.

That would also allow you to share the implementation of both `writeAndValidate` versions, as was done in the other cases.

Reply:
Good point. I implemented the validate across schemas but you are right you can do using existing methods. Let me change.

Reply:
I updated the private method to support `expectedSchema` (optionally) and compare with this schema when provided. Thanks to that, the private `writeAndValidate()` is used in `writeAndValidate(writeSchema, expectedSchema)` (for default values).